django/applications/catmaid/static/js/widgets/tag-table.js

```diff
@@ -436,6 +402,49 @@
     });
   };
 
+  TagTable.prototype.refreshDataAndRedraw = function() {
+    var procSelector = $(`#${this.idPrefix}datatable_processing`);
+
+    procSelector.show();
+
+    for (var key of Object.keys(responseCache)) {
+      delete responseCache[key];
+    }
+    this.resultSkeletons.clear();
+
+    var self = this;
+
+    CATMAID.fetch(project.id + '/labels/stats', 'GET')
+      .then(function(json) {
+        responseCache = json.reduce(function(obj, arr) {
+          var labelID = arr[0];
+          var labelName = arr[1];
+          var skelID = arr[2];
+          var nodeID = arr[3];
+
+          if (!(labelName in obj)) {
+            obj[labelName] = {
+              'labelIDs': new Set(),
+              'skelIDs': new Set(),
+              'nodeIDs': new Map(),
+              'checked': false
+            };
+          }
+
+          obj[labelName].labelIDs.add(labelID);
+          obj[labelName].skelIDs.add(skelID);
+          obj[labelName].nodeIDs.set(nodeID, skelID);
+
+          return obj;
+        }, {});
+
+        self.constrainSkelsAndRedraw();
+
+        procSelector.hide();
```

Comment:
In case of error, the `procSelector` won't be hidden anymore. You could chain a `catch()` call after `then()` to make the processes indicator hide in this case. In there, it would also make sense to call `CATMAID.handleError()` with the function's first argument to have it fail silently.